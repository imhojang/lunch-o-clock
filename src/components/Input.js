import React from 'react';

const Input = (props) => {
  return (
    <form onSubmit={(e) => props.handleSubmit(e)}>
      <input
        type='text'
        value={props.value}
        onChange={(e) => props.handleChange(e)}
      />
      <button>Submit</button>
    </form>
  );
};

export default Input;
