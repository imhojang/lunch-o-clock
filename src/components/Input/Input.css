.input-container {
  margin-bottom: 10px;
}

.input-container > input {
  width: 100%;
  border: 1px solid rgba(0, 0, 0, 0.2);
  border-radius: 5px;
  box-shadow: inset 1px 1px 0 rgba(0, 0, 0, 0.1);
}

.input-container > input:focus {
  border: 1px solid #ffd32e;
  box-shadow: 0 0 10px rgba(255, 255, 0, 0.1);
  outline: none;
}
