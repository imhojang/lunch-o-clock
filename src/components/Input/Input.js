import React from 'react'
import PropTypes from 'prop-types'
import './Input.css'

const Input = ({ value, handleSubmit, handleChange, placeholderText }) => {
  return (
    <form className='input-container' onSubmit={handleSubmit}>
      <input
        type='text'
        value={value}
        onChange={(e) => handleChange(e.target.value)}
        placeholder={placeholderText}
      />
    </form>
  )
}

Input.propTypes = {
  value: PropTypes.string,
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func,
  placeholderText: PropTypes.string
}

export default Input
