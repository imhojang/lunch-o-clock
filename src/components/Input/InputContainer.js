import React from 'react'
import PropTypes from 'prop-types'
import Input from '.'
import { isDuplicateName, removeExtraWhiteSpace } from '../../utils'

class InputContainer extends React.Component {
  constructor (props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  static propTypes = {
    addToList: PropTypes.func.isRequired,
    list: PropTypes.array.isRequired,
    updateInputValue: PropTypes.func.isRequired,
    inputValue: PropTypes.string.isRequired
  };

  handleSubmit (e) {
    const currentValue = removeExtraWhiteSpace(this.props.inputValue)
    const people = this.props.list
    if (isDuplicateName(currentValue, people)) {
      alert(`${currentValue} already exists! Please type in a different name!`)
    } else {
      const addPerson = this.props.addToList
      addPerson(currentValue)
    }
    this.props.updateInputValue('')
    e.preventDefault()
  }

  render () {
    const placeholderText = 'Type and press Enter key to add to list'
    
    return (
      <Input
        handleSubmit={this.handleSubmit}
        handleChange={this.props.updateInputValue}
        value={this.props.inputValue}
        placeholderText={placeholderText}
      />
    )
  }
}

export default InputContainer
