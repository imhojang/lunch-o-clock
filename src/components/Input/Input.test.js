import React from 'react'
import { mount, shallow } from 'enzyme'
import Input from './Input'

describe('Input structure', () => {
  let input

  it('renders correctly', () => {
    input = shallow(<Input />)
  })

  it('matches snapshot', () => {
    expect(input).toMatchSnapshot()
  })

  it('renders form element correctly', () => {
    expect(input.find('form')).toHaveLength(1)
  })

  it('renders input element correctly', () => {
    expect(input.find('input')).toHaveLength(1)
  })
})

describe('Input behavior', () => {
  let input
  let value
  let handleSubmit
  let handleChange

  beforeEach(() => {
    value = 'test'
    handleSubmit = jest.fn()
    handleChange = jest.fn()
    input = mount(
      <Input
        value={value}
        handleSubmit={handleSubmit}
        handleChange={handleChange}
      />
    )
  })

  it('props are passed down correctly', () => {
    expect(input.props().value).toBe(value)
    expect(input.props().handleChange).toBe(handleChange)
    expect(input.props().handleSubmit).toBe(handleSubmit)
  })

  it('form submission calls handleSubmit correctly', () => {
    const form = input.find('form')
    form.simulate('submit')
    expect(handleSubmit).toHaveBeenCalled()
  })

  it('on input change calls handleChange correctly', () => {
    const inputElem = input.find('input')
    inputElem.simulate('change', { value: 'a' })
    expect(handleChange).toHaveBeenCalled()
  })
})
