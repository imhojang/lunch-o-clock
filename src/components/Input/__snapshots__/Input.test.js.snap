// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input structure matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        onChange={undefined}
        placeholder="Type and press Enter key to add to list"
        type="text"
        value={undefined}
/>,
      "className": "input-container",
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "onChange": undefined,
        "placeholder": "Type and press Enter key to add to list",
        "type": "text",
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          onChange={undefined}
          placeholder="Type and press Enter key to add to list"
          type="text"
          value={undefined}
/>,
        "className": "input-container",
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": undefined,
          "placeholder": "Type and press Enter key to add to list",
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
